int8 freq
---
bool ack
